### ISSUE TITLE
Protection value of clothes seems overflowed

### ISSUE THREAD

**Issue by DLaboratory** [CleverRaven/Cataclysm-DDA#17791]
After a long period of playing (about two hours), when I was reading the description of leggings dropped along with a zombie corpse, I suddenly found the protection values are abnormally huge, seemed they were overflowed or corrupted or some other bad things happened. Then I opened the _with for an item_ debug menu, and found all kinds of clothes in `data\json\items\armor.json`, like pants, shirts, cleats, etc., were bugged.

I restarted the program, then everything went back normal. I think there is potential problem in the source code, but I don't know how to reproduce this.

EDIT: Uploaded English screen capture
![1](https://cloud.githubusercontent.com/assets/5671495/17129440/ba9c3a98-5345-11e6-8a95-23086e00161e.png)

----

![1](https://cloud.githubusercontent.com/assets/5671495/17082716/7ec5555e-51ba-11e6-916b-c5fbe65b5da5.png)
<del>Sorry forgot to switch language when taking the screen capture. The two values are bash protection and cut protection.</del>

**Comment by mugling** [comment]
You're compiling with MSVC? This may be identifying undefined behavior in the codebase that has otherwise remained hidden?

**Comment by DLaboratory** [comment]
Yes I compiled it with MSVC. Now I'm encountering this again, investigating in the debugger.

**Comment by DLaboratory** [comment]
```C++
const int eff_damage = to_self ? std::min( damage(), 0 ) : std::max( damage(), 0 );
eff_thickness = std::max( 1, get_thickness() - eff_damage );
```

In `item.cpp:3025`, I found variable `eff_damage` was incredibly small negative number (like -223180016), then caused `eff_thickness` to be a very large positive number. Finally bash protection was calculated by

```C++
return lround((resist * eff_thickness * adjustment) + l_padding + k_padding);
```

I checked `resist`=1, `adjustment`=1.5, `l_padding` and `k_padding` were both zero, these four variables were all correct, the wrong `eff_thickness`  led the final abnormal result.

Then I evaluated the value of what `eff_damage` should be by the same way

```C++
debugmsg("eval = %d\n", to_self ? std::min(damage(), 0) : std::max(damage(), 0));
```

But the result was correct(=1). Maybe I'd run into a compiler bug of MSVC?
